app/utils.py

```diff
@@ -29,3 +33,61 @@ def parse_phone_number(phone_number):
         stripped = '1' + stripped
     stripped = '+' + stripped
     return stripped
+
+
+def parse_to_db(db, filename):
+    viewport_default = '39.861204,-75.310357|40.138932,-74.928582'
+    vehicle_id_index = 8
+    license_plate_index = 9
+    location_index = 4
+    date_index = 0
+    agency_index = 6
+    picture_index = 13
+    description_index = 11
+    with open(filename, 'rb') as file:
+        reader = csv.reader(file)
+        columns = reader.next()
+        for row in reader:
+            address_text = row[location_index]
+            # Viewport-biased geocoding using Google API
+            url = "https://maps.googleapis.com/maps/api/geocode/json"
+            payload = {'address': address_text, 'bounds': viewport_default}
+            r = requests.get(url, params=payload)
+            coordinates = r.json()['results'][0]['geometry']['location']
+            loc = Location(
+                latitude=coordinates['lat'],
+                longitude=coordinates['lng'],
+                original_user_text=address_text)
+            db.session.add(loc)
+            date_format = "%m/%d/%Y %H:%M"
+            time1 = datetime.datetime.strptime(row[date_index], date_format)
+            time2 = datetime.datetime.strptime(row[date_index+1], date_format)
+            # Assign correct agency id
+            agency_name = row[agency_index].rstrip().upper()
+            if agency_name == 'OTHER':
+                agency_name = row[agency_index + 1].rstrip().upper()
+            a = Agency.query.filter_by(name=agency_name).first()
+            if a is None:
+                a = Agency(name=agency_name)
```

Comment:
This should be done by line 66 or 68, but we can change to your case-insensitive method.